=====================
Interactive Shiny App
=====================

Introduction
============
Intervene Shiny App provides an interactive interface for intersection and effective visualization of gene or genomic region sets. Currently, Shiny app does not acccept genomic regions as input, but the text files generated by Interve's command line interface can be easily uploaded to further explore and customize the plots in an interactive way. Intervene has three modules: venn to generate Venn diagrams of up-to 6 sets, upset to generate UpSet plots of more than 3 sets and pairwise to compute and visualize pariwise intersections as clustered heatmap.

.. figure:: img/shinyapp_main.png
   :alt: Interactive Shiny App


Venn module
===========
Intervene's venn module provides up-to 6-way classical, Chow-Ruskey and Edwards’ Euler/Venn diagrams to visualize the intersections of genomic regions or lists.

Usage instructions
------------------
To use this venn module, you can upload a correctly formatted csv/text file, with lists of names. Each column represents a set, and each row represents an element (names/gene/SNPs).

Before uploading the file, choose the correct separator, wheather the names in each column are seperated by a ' , ' choose comma, by a ' ; ' choose semicolon, or by tabs choose tab.

Header names (first row) will be used as set names.

Intervene uses the Vennerable R package to generate different Venn diagrams.


UpSet module
============
Intervene’s UpSet modules can be used to visualize the intersection of multiple genomic region sets using UpSet plots.

Usage instructions
------------------

To use this module you can upload a correctly formatted .csv or text file, encoded in binary. Before uploading the file, choose the correct separator, wheather the names in each column are seperated by a ' , ' choose comma, by a ' ; ' choose semicolon, or by tabs choose tab. Header names (first row) will be used as set names.

UpSet module takes three types of inputs.

**List type data**

List data is a correctly formatted csv/text file, with lists of names. Each column represents a set, and each row represents an element (names/gene/SNPs). Header names (first row) will be used as set names.

**Binary type data**

In the binary input file each column represents a set, and each row represents an element. If a names is in the set then it is represented as a 1, else it is represented as a 0.

**Combination/expression type data**

Combination/expression type data is the possible combinations of set intersections. User can copy/past the combinations of intersection from the Intervene commnad line interface. For example;

H3K4me2&H3K4me3=2216, H3K4me2&H3K4me3&H3K27me3=6777, H3K27me3=5909, H3K4me3&H3K27me3=307, H3K4me3=256, H3K4me2&H3K27me3=3852, H3K4me2=15676, H3K27ac&H3K4me2&H3K4me3&H3K27me3=7235, H3K27ac&H3K4me2&H3K4me3=17505, H3K27ac&H3K4me2=12011, H3K27ac&H3K4me2&H3K27me3=1698, H3K27ac&H3K4me3=473, H3K27ac&H3K4me3&H3K27me3=295, H3K27ac&H3K27me3=1490, H3K27ac=15021

Intervene uses the UpSetR R package for visualization.

Pairwise module
===============
Intervene’s pairwise module provides several styles of heatmaps and clustering approaches to customize the heatmaps.

Usage instructions
------------------
To use pairwise module, you can upload a pairwise matrix file in .csv/txt format. Each column and row represents pairwise fraction of overlap/count etc between different names/genomic region sets.

Before uploading the file, choose the correct separator, wheather the matrix file is seperated by a ' , ' choose comma, by a ' ; ' choose semicolon, or by tabs choose tab.

Pairwise module takes input of two types:

**List type data**

List data is a correctly formatted csv/text file, with lists of names. Each column represents a set, and each row represents an element (names/gene/SNPs). Header names (first row) will be used as set names.

**Pairwise matrix data**

A pairwise matrix type data is a matrix of size NxN (all pairwise combinations) with values as number/fraction of overlap between two corresponding sets. For genomic region sets user can use the commpnad line interface of Intervene and upload the generated matrix here as matrix type.

For example here is the demo data generated by Intervene's command line interfacce for super-enhancers(SEs) of different cell/tissue-types from dbSUPER.

Intervene uses the Corrplot and plotly R packages to plot heatmap

Availability
============
The Intervene Shiny App is freely available at:

> https://intervene.shinyapps.io/intervene/
> https://asntech.shinyapps.io/intervene